src/contracts/libraries/GPv2Transfer.sol

```diff
@@ -81,39 +77,68 @@ library GPv2Transfer {
         Data[] calldata transfers,
         address recipient
     ) internal {
-        // NOTE: Pre-allocate an array of vault balance tranfers large enough to
-        // hold all transfers. This allows us to efficiently batch internal
-        // balance transfers into a single Vault call.
+        uint256 transferCount = transfers.length;
+
+        // NOTE: Allocate an array of Vault balance transfers large enough
```

Comment:
Would you mind benchmarking the gas savings of this trick over simply using two arrays of transfers.length size?

Reply:
Sure. I just want slice semantics from Rust back :sob:

Reply:
I think I'll do a naive implementation for this PR and do the fancy one in a separe PR so we can discuss there.

Reply:
Turns out its not so significant:
```
=== Settlement Gas Benchmarks ===
--------------+--------------+--------------+--------------+--------------
       tokens |       trades | interactions |      refunds |          gas 
--------------+--------------+--------------+--------------+--------------
            2 |           10 |            0 |            0 |       858182  (change: 167 / trade)
            3 |           10 |            0 |            0 |       862914  (change: 167 / trade)
            4 |           10 |            0 |            0 |       867658  (change: 166 / trade)
            5 |           10 |            0 |            0 |       872378  (change: 169 / trade)
            6 |           10 |            0 |            0 |       877110  (change: 165 / trade)
            7 |           10 |            0 |            0 |       881794  (change: 166 / trade)
            8 |           10 |            0 |            0 |       886466  (change: 162 / trade)
            8 |           20 |            0 |            0 |      1673472  (change: 169 / trade)
            8 |           30 |            0 |            0 |      2439165  (change: 174 / trade)
            8 |           40 |            0 |            0 |      3205734  (change: 177 / trade)
            8 |           50 |            0 |            0 |      3971932  (change: 180 / trade)
            8 |           60 |            0 |            0 |      4739179  (change: 182 / trade)
            8 |           70 |            0 |            0 |      5506786  (change: 186 / trade)
            8 |           80 |            0 |            0 |      6274635  (change: 188 / trade)
            2 |           10 |            1 |            0 |       929546  (change: 168 / trade)
            2 |           10 |            2 |            0 |       976162  (change: 162 / trade)
            2 |           10 |            3 |            0 |      1022766  (change: 161 / trade)
            2 |           10 |            4 |            0 |      1069394  (change: 161 / trade)
            2 |           10 |            5 |            0 |      1116022  (change: 163 / trade)
            2 |           10 |            6 |            0 |      1162698  (change: 172 / trade)
            2 |           10 |            7 |            0 |      1209290  (change: 166 / trade)
            2 |           10 |            8 |            0 |      1255930  (change: 167 / trade)
            2 |           50 |            0 |           10 |      3642398  (change: 182 / trade)
            2 |           50 |            0 |           15 |      3601632  (change: 180 / trade)
            2 |           50 |            0 |           20 |      3560940  (change: 181 / trade)
            2 |           50 |            0 |           25 |      3520141  (change: 182 / trade)
            2 |           50 |            0 |           30 |      3479367  (change: 182 / trade)
            2 |           50 |            0 |           35 |      3438630  (change: 181 / trade)
            2 |           50 |            0 |           40 |      3397834  (change: 179 / trade)
            2 |           50 |            0 |           45 |      3357134  (change: 183 / trade)
            2 |           50 |            0 |           50 |      3316352  (change: 180 / trade)
            2 |            2 |            0 |            0 |       212332  (change: 136 / trade)
            2 |            1 |            1 |            0 |       202895  (change: 90 / trade)
           10 |          100 |           10 |           20 |      7634217  (change: 193 / trade)
```

I think we can leave it out.